lib/modules.nix

```
@@ -773,14 +886,15 @@ rec {
         visible = false;
         apply = x: throw "The option `${showOption optionName}' can no longer be used since it's been removed. ${replacementInstructions}";
       });
-      config.assertions =
-        let opt = getAttrFromPath optionName options; in [{
-          assertion = !opt.isDefined;
+      config._module.checks =
+        let opt = getAttrFromPath optionName options; in {
+        ${showOption optionName} = lib.mkIf opt.isDefined {
```

```suggestion
        ${"removed-" + showOption optionName} = lib.mkIf opt.isDefined {
```

Avoid a bad, unexpected error message in case someone named a domain specific check the same as an option they're removing.